lib/rules/no-expose-after-await.js

```diff
@@ -56,147 +56,195 @@ module.exports = {
-          return
-        }
-        setupScope.afterAwait = true
-      },
-      /** @param {CallExpression} node */
-      CallExpression(node) {
-        if (!scopeStack) {
-          return
-        }
-        const setupScope = setupScopes.get(scopeStack.scopeNode)
-        if (
-          !setupScope ||
-          !setupScope.afterAwait ||
-          !utils.inRange(setupScope.range, node)
-        ) {
-          return
-        }
-        const { contextReferenceIds, exposeReferenceIds } = setupScope
-        if (
-          node.callee.type === 'Identifier' &&
-          exposeReferenceIds.has(node.callee)
-        ) {
-          // setup(props,{expose}) {expose()}
-          context.report({
-            node,
-            messageId: 'forbidden'
-          })
-        } else {
-          const expose = getCalleeMemberNode(node)
+          const { isContextReferenceId, isExposeReferenceId } = setupScope
           if (
-            expose &&
-            expose.name === 'expose' &&
-            expose.member.object.type === 'Identifier' &&
-            contextReferenceIds.has(expose.member.object)
+            node.callee.type === 'Identifier' &&
+            isExposeReferenceId(node.callee, node)
           ) {
-            // setup(props,context) {context.emit()}
+            // setup(props,{expose}) {expose()}
             context.report({
               node,
-              messageId: 'forbidden'
+              messageId: 'forbidden',
+              data: {
+                name: node.callee.name
+              }
             })
+          } else {
+            const expose = getCalleeMemberNode(node)
+            if (
+              expose &&
+              expose.name === 'expose' &&
+              expose.member.object.type === 'Identifier' &&
+              isContextReferenceId(expose.member.object)
+            ) {
+              // setup(props,context) {context.emit()}
+              context.report({
+                node,
+                messageId: 'forbidden',
+                data: {
+                  name: expose.name
+                }
+              })
+            }
           }
         }
       },
-      onSetupFunctionExit(node) {
-        setupScopes.delete(node)
-      }
-    })
+      (() => {
+        const scriptSetup = utils.getScriptSetupElement(context)
+        if (!scriptSetup) {
+          return {}
+        }
+        return {
+          /**
+           * @param {Program} node
+           */
+          Program(node) {
+            context
+              .getScope()
+              .references.find((ref) => ref.identifier.name === 'defineExpose')
+            setupScopes.set(node, {
+              afterAwait: false,
+              range: scriptSetup.range,
+              isExposeReferenceId: (_id, callNode) => {
+                return (
+                  callNode.parent.type === 'ExpressionStatement' &&
+                  callNode.parent.parent === node
+                )
+              },
```

Comment:
A bit shorter (though it may be worth to check how Prettier would format it):

```suggestion
              isExposeReferenceId: (_id, callNode) => (
                callNode.parent.type === 'ExpressionStatement' &&
                callNode.parent.parent === node
              ),
```